packages/extensions/structure/src/sdk.ts

```diff
@@ -0,0 +1,144 @@
+import type Dexie from 'dexie'
+import type {
+	Attribute,
+	DatabaseRecord,
+	QualifiedAttribute,
+} from '../node_modules/@septkit/fileio/dist/common/common.types'
+
+export function useSDK(db: Dexie) {
+	return {
+		findChildRecrods,
+		templateify,
+		ensureRelationship,
+		addRecord,
+	}
+
+	/**
+	 *
+	 * @param record the added record
+	 */
+	async function addRecord(record: DatabaseRecord): Promise<DatabaseRecord> {
+		try {
+			const addedId = await db.table<DatabaseRecord>(record.tagName).add(record)
+			const addedRecord = await db.table<DatabaseRecord>(record.tagName).get(addedId)
+			if (!addedRecord) {
+				const errMsg = { msg: 'could not find added record', table: record.tagName, addedId }
+				console.error(errMsg)
+				throw new Error(JSON.stringify(errMsg))
+			}
+
+			return addedRecord
+		} catch (err) {
+			const errMsg = {
+				msg: 'could not add record',
+				db: db.name,
+				table: record.tagName,
+				record,
+				err,
+			}
+			console.error(errMsg)
+			throw new Error(JSON.stringify(errMsg))
+		}
+	}
+
+	async function ensureRelationship(parent: DatabaseRecord, child: DatabaseRecord) {
+		const parentHasChild = parent.children?.some(
+			(c) => c.id === child.id && c.tagName === c.tagName,
+		)
+		const childHasParent =
+			child.parent?.id === parent.id && child.parent?.tagName === parent.tagName
+
+		if (!parentHasChild) {
+			parent.children?.push({
+				id: child.id,
+				tagName: child.tagName,
+			})
+			await db.table(parent.tagName).update(parent.id, parent)
+		}
+
+		if (!childHasParent) {
+			child.parent = {
+				id: parent.id,
+				tagName: parent.tagName,
+			}
+			await db.table(child.tagName).update(child.id, child)
+		}
+	}
+
+	// TODO: fsdReference with file name
+	function templateify(record: DatabaseRecord) {
+		const uuid = extractUuid(record)
+		if (!uuid) {
+			console.error('record does not have a "uuid" to move to "templateUUID"', record)
+			throw new Error('record does not have a "uuid" to move to "templateUUID"')
+		}
+		record.attributes?.push({ name: 'templateUuid', value: uuid })
+		addNewUuid(record)
+	}
+
+	function extractUuid(record: DatabaseRecord): string | undefined {
+		const uuidAttr = record.attributes?.find((atr) => atr.name === 'uuid')
+		if (!uuidAttr) {
+			return undefined
+		}
+
+		return uuidAttr.value
+	}
+	function addNewUuid(record: DatabaseRecord) {
+		if (!record.attributes) {
+			record.attributes = []
+		}
+
+		const newUuid = crypto.randomUUID()
+		const uuidAttrIndex = record.attributes.findIndex((atr) => atr.name === 'uuid')
+
+		const hasUuid = uuidAttrIndex < 0
+
+		if (!hasUuid) {
+			record.attributes[uuidAttrIndex].value = newUuid
+		} else {
+			record.attributes.push({ name: 'uuid', value: newUuid })
+		}
```

Comment:
The variable `hasUuid` has the inverted value and is used the other way around? Was this intentional? I would prefer:
```suggestion
		const hasUuid = uuidAttrIndex >= 0

		if (hasUuid) {
			record.attributes[uuidAttrIndex].value = newUuid
		} else {
			record.attributes.push({ name: 'uuid', value: newUuid })
		}
```

Reply:
I would also prefer the correct way :D